api/src/API/Controllers/RacesController.cs

```diff
@@ -40,17 +43,78 @@ public async Task<IActionResult> GetAll()
             return Ok(result);
         }
 
+        [AllowAnonymous]
+        [HttpGet("public")]
+        public async Task<IActionResult> GetAllPublic()
+        {
+            RaceContainerClient container = containerProvider.RaceContainer;
+            IEnumerable<Race> result = await container.GetAllAsync();
+            return Ok(result);
+        }
+
         [HttpPost]
         public async Task<IActionResult> Create(Race race)
         {
             RaceContainerClient container = containerProvider.RaceContainer;
-            if (race.EventId == Guid.Empty)
+            if (!(await InitAndVerifyRace(race, container)))
             {
-                race.EventId = Guid.NewGuid();
+                return BadRequest();
             }
 
             await container.AddOneAsync(race);
             return CreatedAtAction(nameof(Create), new { id = race.Id }, race);
         }
+
+        [AllowAnonymous]
+        [HttpPost("public")]
+        public async Task<IActionResult> CreatePublic(Race race)
+        {
+            RaceContainerClient container = containerProvider.RaceContainer;
+            race.IsPublic = false;
```

Comment:
I would move this after the check below, since we shouldn't modify anything in the request before it's verified

Reply:
I'm not sure it makes a difference in the case of returning a BadRequest but sure.